src/fparser/splitline.py

```diff
@@ -176,313 +182,122 @@ def splitquote(line, stopchar=None, lower=False, quotechars = '"\''):
-    Handles splitting lines with incomplete string blocks.
-    """
-    def __init__(self, line,
-                 quotechar = None,
-                 lower=False,
-                 ):
-        self.fifo_line = [c for c in line]
-        self.fifo_line.reverse()
-        self.quotechar = quotechar
-        self.lower = lower
-
-    def split2(self):
-        """
-        Split line until the first start of a string.
-        """
-        try:
-            item1 = self.get_item()
-        except StopIteration:
-            return '',''
-        i = len(item1)
-        l = self.fifo_line[:]
-        l.reverse()
-        item2 = ''.join(l)
-        return item1,item2
-
-    def get_item(self):
-        fifo_pop = self.fifo_line.pop
-        try:
-            char = fifo_pop()
-        except IndexError:
-            raise StopIteration
-        fifo_append = self.fifo_line.append
-        quotechar = self.quotechar
-        l = []
-        l_append = l.append
-
-        nofslashes = 0
-        if quotechar is None:
-            # search for string start
-            while 1:
-                if char in '"\'' and not nofslashes % 2:
-                    self.quotechar = char
-                    fifo_append(char)
-                    break
-                if char=='\\':
-                    nofslashes += 1
-                else:
-                    nofslashes = 0
-                l_append(char)
-                try:
-                    char = fifo_pop()
-                except IndexError:
-                    break
-            item = ''.join(l)
-            if self.lower: item = item.lower()
-            return item
-
-        if char==quotechar:
-            # string starts with quotechar
-            l_append(char)
-            try:
-                char = fifo_pop()
-            except IndexError:
-                return String(''.join(l))
-        # else continued string
-        while 1:
-            if char==quotechar and not nofslashes % 2:
-                l_append(char)
-                self.quotechar = None
-                break
-            if char=='\\':
-                nofslashes += 1
-            else:
-                nofslashes = 0
-            l_append(char)
-            try:
-                char = fifo_pop()
-            except IndexError:
-                break
-        return String(''.join(l))
 
-def splitparen(line,paren='()'):
+def splitparen(line, paren_open="([", paren_close=")]"):
     """
-    Fast LineSplitterParen.
+    Splits a line into top-level parenthesis and not-parenthesised
+    parts. E.g.: "a( (1+2)*3) = b(x)" becomes:
+    ["a", "( (1+2)*3)", " = b", "(x)"]
+    :param str line: the string to split.
+    :param str paren_open: The characters that define an open parentheses.
+    :param str paren_close: The characters that define a closing parentheses.
+    The paren_open and paren_close strings must be matched in order:
+    paren_open[x] is closed by paren_close[x].
```

Comment:
Great but please also add:

    :return: List of parenthesised and not-parenthesised parts
    :rtype: list of str